### PR TITLE
Add malloc size checks

### PR DESCRIPTION
**Pull request by ericwol-msft**

### CODE REVIEW

**Review by RLeclair — APPROVED** [review]
LGTM!

File: src/constbuffer.c [hunk 1]
```diff
@@ -33,7 +33,18 @@ static CONSTBUFFER_HANDLE CONSTBUFFER_Create_Internal(const unsigned char* sourc
     CONSTBUFFER_HANDLE result;
     /*Codes_SRS_CONSTBUFFER_02_005: [The non-NULL handle returned by CONSTBUFFER_Create shall have its ref count set to "1".]*/
     /*Codes_SRS_CONSTBUFFER_02_010: [The non-NULL handle returned by CONSTBUFFER_CreateFromBuffer shall have its ref count set to "1".]*/
-    result = (CONSTBUFFER_HANDLE)calloc(1, (sizeof(CONSTBUFFER_HANDLE_DATA) + size));
+    size_t malloc_size = sizeof(CONSTBUFFER_HANDLE_DATA) + size;
+    if (malloc_size < size)
```
Comment (vaavva) [inline src/constbuffer.c:37]: same here as first comment

**Review by vaavva — APPROVED** [review]

File: adapters/httpapi_compact.c [hunk 2]
```diff
@@ -431,7 +432,18 @@ static void on_bytes_received(void* context, const unsigned char* buffer, size_t
         else
         {
             /* Here we got some bytes so we'll buffer them so the receive functions can consumer it */
-            new_received_bytes = (unsigned char*)realloc(http_instance->received_bytes, http_instance->received_bytes_count + size);
+            size_t malloc_size = http_instance->received_bytes_count + size;
+            if (malloc_size < size)
```
[the remark this answers is not in the document]
Comment (ericwol-msft) [reply adapters/httpapi_compact.c:436]: I validated this, see teams chat

File: adapters/httpapi_winhttp.c [hunk 3]
```diff
@@ -143,7 +156,8 @@ static HTTPAPI_RESULT ConstructHeadersString(HTTP_HEADERS_HANDLE httpHeadersHand
                 result = HTTPAPI_STRING_PROCESSING_ERROR;
                 LogError("MultiByteToWideChar failed, GetLastError=0x%08x (result = %" PRI_MU_ENUM ")", GetLastError(), MU_ENUM_VALUE(HTTPAPI_RESULT, result));
             }
-            else if ((*httpHeaders = (wchar_t*)malloc((requiredCharactersForHeaders + 1) * sizeof(wchar_t))) == NULL)
+            else if ((requiredCharactersForHeaders + 1) == 0 ||  // int overflow check
+                (*httpHeaders = (wchar_t*)malloc((requiredCharactersForHeaders + 1) * sizeof(wchar_t))) == NULL)
```
[the remark this answers is not in the document]
Comment (ericwol-msft) [reply adapters/httpapi_winhttp.c:160]: yes, using wchar_t :(

File: src/constbuffer.c (hunk 1 above)
Comment (ericwol-msft) [reply src/constbuffer.c:37]: this is valid

**Pull request merged**